internal/proxy/task_delete.go

```diff
@@ -446,24 +472,8 @@ func (dr *deleteRunner) produce(ctx context.Context, primaryKeys *schemapb.IDs)
 func (dr *deleteRunner) getStreamingQueryAndDelteFunc(plan *planpb.PlanNode) executeFunc {
 	return func(ctx context.Context, nodeID int64, qn types.QueryNodeClient, channel string) error {
 		var partitionIDs []int64
-
-		// optimize query when partitionKey on
-		if dr.partitionKeyMode {
-			expr, err := exprutil.ParseExprFromPlan(plan)
-			if err != nil {
-				return err
-			}
-			partitionKeys := exprutil.ParseKeys(expr, exprutil.PartitionKey)
-			hashedPartitionNames, err := assignPartitionKeys(ctx, dr.req.GetDbName(), dr.req.GetCollectionName(), partitionKeys)
-			if err != nil {
-				return err
-			}
-			partitionIDs, err = getPartitionIDs(ctx, dr.req.GetDbName(), dr.req.GetCollectionName(), hashedPartitionNames)
-			if err != nil {
-				return err
-			}
-		} else if dr.partitionID != common.InvalidFieldID {
-			partitionIDs = []int64{dr.partitionID}
+		if len(dr.partitionIDs) > 0 {
+			partitionIDs = dr.partitionIDs
```

Comment:
`partitionIDs` can be nil in the newer code path, is it safe for future invocations?

Reply:
If I understands you correctlly, nil is safe to len(), and will always check len() != 0 before indexing